app/controllers/EHealthVDCController.scala

```diff
@@ -1,144 +1,163 @@
+class EHealthVDCController @Inject() (config: Configuration, initService: Init, ws: WSClient) extends InjectedController {
+  private val LOGGER = LoggerFactory.getLogger("EHealthVDCController")
+
+  def sendRequestToEnforcmentEngine (query:String, purpose: String, requesterId: String, url:String): String = {
+    val data = Json.obj(
+      "query" -> query,
+      "purpose" -> purpose,
+      "access" -> "read",
+      "requester" -> "",
+      "blueprintId" -> "",
+      "requesterId" -> requesterId
+    )
+    val futureResponse = ws.url(url).addHttpHeaders("Content-Type" -> "application/json")
+      .addHttpHeaders("Accept" -> "application/json").withRequestTimeout(Duration.Inf).post(data)
+
+    val res = Await.result(futureResponse, 100 seconds)
+    res.body[String]
   }
 
   @ApiOperation(nickname = "getAllValuesForBloodTestComponent",
     value = "Get timeseries of patient's blood test component",
     notes =  "This method returns the collected values for a specific blood test component of a patient (identified by his SSN), to be used by medical doctors",
-    response = classOf[models.Patient], responseContainer = "List", httpMethod = "GET")
+    response = classOf[models.BloodTestComponents], responseContainer = "List", httpMethod = "GET")
   @ApiResponses(Array(
     new ApiResponse(code = 404, message = "Patient not found")))
-  def getTestValues(@ApiParam(value = "SSN", required = true,
-    allowMultiple = false) socialId: String,
+  def getAllValuesForBloodTestComponent(@ApiParam(value = "SSN", required = true, allowMultiple = false) socialId: String,
+                    @ApiParam(value = "requesterId", required = true,
+                      allowMultiple = false) requesterId: String,
                     @ApiParam(value = "component", required = true,
-                      allowMultiple = false) testType: String): Action[AnyContent] = {
-    Action.async {
+                      allowMultiple = false) testType: String)= Action.async {
+    implicit request =>
       val spark = initService.getSparkSessionInstance
-      readData(spark)
-      val query = "select patientId, date, %s.value as %s from joined where socialId='%s'".format(testType, testType, socialId)
-      val patientBloodTestsDF = spark.sql(query)
-      patientBloodTestsDF.limit(10).show(false)
-      patientBloodTestsDF.printSchema
-       patientBloodTestsDF.explain(true)
-
-      val rawJson = patientBloodTestsDF.limit(10).toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
-    }
+      val patientSSN = socialId
+      var origtestType = testType
+      var newTestType: String = null
+      if (origtestType.equals("cholesterol")) {
+        newTestType = "cholesterol_total_value"
+      } else {
+        newTestType = "%s_value".format(origtestType)
+      }
+      val queryToEngine = "SELECT patientId, date, %s FROM blood_tests".format(newTestType)
+
+      if (config.has("policy.enforcement.play.url")) {
```

Comment:
sendRequestToEnforcmentEngine should do this logic. It should receive a query with all the params , and it should return a rewritten query with all the additional tables needed to instantiate.
Then there should be another method that does the actual query with instantiation of all the needed tables.